astropy/config/configuration.py

```diff
@@ -570,6 +574,57 @@ def get_config(packageormod=None, reload=False, rootname=None):
         return cobj
 
 
+def generate_config(filename=None):
```

Comment:
Other packages use the configuration system too, so should be more explicit in the name that this won't work for them. Or find a way to let user customize the list of modules.
```suggestion
def generate_astropy_config(filename=None):
```